frontend/src/components/App.js

```diff
@@ -4,51 +4,115 @@ import './App.css';
 import Grid from "@material-ui/core/Grid";
 import DatePicker from "./DatePicker";
 import CalendarLink from "./CalendarLink";
-
-//todo get from API
-const CALENDAR_LINK = 'https://calendar.google.com/calendar?cid=MXQ0NDRob24wdTE1cGkxOWlyYzUxaTgxb3NAZ3JvdXAuY2FsZW5kYXIuZ29vZ2xlLmNvbQ';
-const CALENDAR_ID = '1t444hon0u15pi19irc51i81os@group.calendar.google.com';
+import CalendarSelect from "./CalendarSelect";
 
 //todo move to prop file
-const ADDRESS = 'http://192.168.88.254:5000';
-const API = '/api/calendar';
+const ADDRESS = 'http://192.168.88.254:5000/';
+const API = 'api/calendar/';
 
 class App extends React.Component {
     constructor(props) {
         super(props);
         this.state = {
             events: [],
-            start: new Date()
+            calendars:
+                [],
+            start: new Date(),
+            calendar: {}
         }
     }
 
     abortController = new window.AbortController();
 
     componentDidMount() {
-        this.fetchEvents()
+        this.fetchCalendars().then(() => this.fetchEvents());
     }
 
     componentDidUpdate(prevProps, prevState) {
         if (this.state.start !== prevState.start) {
             this.fetchEvents()
         }
+
+        if (this.state.calendar !== prevState.calendar) {
+            this.fetchEvents()
+        }
     }
 
     componentWillUnmount = () => this.abortController.abort();
 
+    async fetchCalendars() {
+        let url = ADDRESS + API;
+
+        let noData = {
+            link: 'nodata',
+            id: 'nodata',
+            name: 'nodata',
+        };
+
+        let a = [];
+        a.push(noData);
+
+        await fetch(url)
+            .then(res => res.json())
+            .then(calendars => {
+                if (calendars.length !== 0) {
+                    this.setState({
+                        calendars: calendars,
+                        calendar: calendars[0],
+                    })
+                } else {
+                    this.setState({
+                        calendar: noData,
+                        calendars: a
+                    });
+                }
+            })
+            .catch(error => {
+                if (error.name === 'AbortError') {
+                    this.setState({
+                        calendar: noData,
+                        calendars: a
+                    });
+                }
+            });
+    }
+
     fetchEvents() {
-        let url = new URL(ADDRESS + API);
-        let params = [['calendarId', CALENDAR_ID], ['start', this.state.start.toISOString()]];
+        let url = new URL(ADDRESS + API + this.state.calendar.id);
+        let params = [['start', this.state.start.toISOString()]];
         url.search = new URLSearchParams(params).toString();
 
+        let noEvents = {
```

Comment:
That's not a good idea to use list item as a placeholder for empty list. We probably can render another component